### PR TITLE
Template update for nf-core/tools version 1.8.dev0

### PR DESCRIPTION
**Pull request by ewels**
Automated `TEMPLATE` branch update for the latest version of tools.

Note that because the refactored sync code is not yet released (or even merged - https://github.com/nf-core/tools/pull/423), some of these changes may be ahead of the latest stable lint tests. This might mean that some stuff fails.

### CODE REVIEW

**Comment by drpatelh** [comment]
@phue Add yourself in the `Credits` section!

**Comment by phue** [comment]
> @phue Add yourself in the `Credits` section!

Thanks, I'll gladly do that as soon as I figured out how to checkout this PR locally :flushed:

**Comment by phue** [comment]
Not sure if that's what I intended to do. I used the web editor thing and then it merged `dev` into `TEMPLATE`, shouldn't it be the other way around?
I admit that I don't fully understand how this TEMPLATE syncing magic works, but anyhow feel free to revert if this was wrong :grimacing:

**Comment by ewels** [comment]
Umm, no it's the other way around - this branch needs to be merged into `dev`. We shouldn't need to add any more commits to the `TEMPLATE` branch. I think.

I don't think that it'll be easy to do through the web editor due to the somewhat complex merge conflicts, plus the fact that some of the boilerplate code _shouldn't_ be present in the actual pipeline.

Docs here: https://nf-co.re/developers/sync#merge-template-into-main-branches

TLDR; I _think_ this is correct, working locally on your fork and assuming that you have the head nf-core fork set up as a remote called `upstream`:

```bash
git fetch upstream
git checkout -b template_merge
git merge upstream/TEMPLATE
# ..fix merge conflicts, edit code, commit changes
git checkout dev
git merge --no-ff template_merge
git push
# open from your fork 'template_merge' branch to upstream dev
```

Does that make sense? ..I get a bit lost with this sometimes too! 😓 

Note that we'll need to revert your commits above first for this to all work I think. Sorry..

**Comment by phue** [comment]
Thanks @ewels it makes more sense now. But this would require a different PR then, coming from my own fork?
I was trying to figure out how to checkout this PR here and then resolve the conflicts.
I failed and instead ended up pushing to `TEMPLATE` directly

**Comment by ewels** [comment]
> But this would require a different PR then, coming from my own fork?

Yes it would 👍 However, if that other PR contains the commits present in this PR, this PR will close / be merged when the other one is merged.

This PR directly from the `TEMPLATE` branch really only makes sense if the template updates are very small / minor and can be merged without modification. I mostly like them because opening the PR highlights the changes so we can see quickly what needs to be done.

If you can check out a new branch on your fork with your changes, then `git reset` / force push the `TEMPLATE` branch again to keep it clean, that would be fab 👍

**Comment by ewels** [comment]
Thank you for reverting this 👍 

Now to see if this magically closes / shows as merged after your other PR is merged. If it doesn't I may need to revisit my git knowledge 😅

**Comment by ewels** [comment]
Nice, it worked! Great work @phue

**Comment by phue** [comment]
image
